feature/checkstyle-integration

0001vrn opened the pull request:
- add required checkstyle configs
- updated poms as per checkstyle configs
- fixed checkstyle violations

Closes #602

0001vrn commented on modelservice/src/test/java/com/expedia/adaptivealerting/modelservice/util/ErrorDetailsTest.java:
```diff
@@ -1,3 +1,18 @@
+ */
 package com.expedia.adaptivealerting.modelservice.util;
 
 public class ErrorDetailsTest {
```
@bibinss Can I remove this file? Since it's empty and of no use.

bibinss replied:
yes please

0001vrn replied:
Sure

bibinss commented on pom.xml:
```diff
@@ -751,6 +758,30 @@
+                </configuration>
+                <dependencies>
+                    <dependency>
+                        <groupId>com.puppycrawl.tools</groupId>
```
what does this dependency do?

0001vrn replied:
With this, we can override the version of the underlying checkstyle library. Otherwise, it is tied to the default checkstyle which may be buggy(or not stable). 

Doing so will give us the flexibility to use the stable version of checkstyle.

0001vrn replied:
Regarding the library itself:
Checkstyle is a tool for checking Java source code for adherence to a Code Standard or set of validation rules (best practices).

More context - https://github.com/checkstyle/checkstyle

bibinss replied:
plugin version is defined as `<version>${maven.checkstyle.plugin.version}</version>` right? my question is do we really need this dependency? I can see the checkstyle getting applies even if I remove this dependency.

0001vrn replied:
Removing this will result in checkstyle giving violations even in Java Annotations.
Let's keep it as it is.

bibinss commented on pom.xml:
```diff
@@ -751,6 +758,30 @@
+                <configuration>
+                    <configLocation>${checkstyle.config.location}</configLocation>
+                    <headerLocation>${checkstyle.header.file}</headerLocation>
+                    <propertyExpansion>samedir=${checkstyle.basedir}</propertyExpansion>
```
why we need this?

0001vrn replied:
Earlier, I was injecting the header file name using this. But later on, I was able to use default variables to refer to the same. 

Looks like we can go ahead without this particular config.

bibinss replied:
ok, lets remove this.

0001vrn replied:
Sure

bibinss commented on build/rules/checkstyle_rules.xml:
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0"?>
```
can we rename this folder as `checkstyle` and keep the file under it

0001vrn replied:
Yeah. Sure

bibinss commented on pom.xml:
```diff
@@ -137,6 +138,12 @@
+        <maven.checkstyle.plugin.version>3.0.0</maven.checkstyle.plugin.version>
+
+        <checkstyle.basedir>${main.basedir}/build/rules</checkstyle.basedir>
+        <checkstyle.config.location>${main.basedir}/build/rules/checkstyle_rules.xml</checkstyle.config.location>
```
once you move your checkstyle rules under `checkstyle` folder, you can define this as `<checkstyle.config.location>checkstyle/checkstyle_rules.xml</checkstyle.config.location>`

0001vrn replied:
Makes sense. Tested it out locally, it works fine.

bibinss commented on pom.xml:
```diff
@@ -137,6 +138,12 @@
+
+        <checkstyle.basedir>${main.basedir}/build/rules</checkstyle.basedir>
+        <checkstyle.config.location>${main.basedir}/build/rules/checkstyle_rules.xml</checkstyle.config.location>
+        <checkstyle.header.file>${main.basedir}/build/rules/license_header</checkstyle.header.file>
```
similar comment as above

0001vrn replied:
Gotcha. Thanks

bibinss commented on kafka/pom.xml:
```diff
@@ -34,6 +34,7 @@
     <description>Kafka interface for Adaptive Alerting.</description>
 
     <properties>
+        <main.basedir>${project.parent.basedir}</main.basedir>
```
you mayn't need this config change. similarly in other module's pom as well.

0001vrn replied:
Understood.

bibinss commented on pom.xml:
```diff
@@ -71,6 +71,7 @@
         </repository>
     </distributionManagement>
     <properties>
+        <main.basedir>${project.basedir}</main.basedir>
```
is this required?

0001vrn replied:
Not anymore, perhaps.

0001vrn commented:
Hi @williewheeler 
I have fixed merge conflicts and fixed checkstyle violations that came due to those changes. 

Please review and merge this PR.

Thanks

williewheeler commented:
@0001vrn Merged -- thank you very much for this submission.
